src/v2/view-builder/views/ov/ResendView.js

```diff
@@ -31,9 +32,23 @@ export default View.extend({
   },
 
   showCalloutWithDelay() {
-    this.showMeTimeout = _.delay(() => {
-      this.$el.removeClass('hide');
-    }, SHOW_RESEND_TIMEOUT);
+    const timeStamp = sessionStorageHelper.getResendTimestamp();
+    if (!timeStamp) {
+      sessionStorageHelper.setResendTimestamp(Date.now());
+    }
+
+    // We keep track of a 'global' timestamp in sessionStorage because if the SIW does a re-render,
+    // we don't want to force the user to wait another 30s again to see the resend link. With this
+    // the user will wait AT MOST 30s until they see the resend link.
+    const start = sessionStorageHelper.getResendTimestamp();
+    this.showMeTimeout = setInterval(() => {
+      const now = Date.now();
+      if (now - start >= SHOW_RESEND_TIMEOUT) {
+        this.$el.removeClass('hide');
+        clearInterval(this.showMeTimeout);
+        sessionStorageHelper.removeResendTimestamp();
+      }      
+    }, 250);
```

Comment:
`showCalloutWithDelay` could go into a base view that these other views extend from

Reply:
This is done now.